core/parachain/pvf/kagome_pvf_worker.hpp

```diff
@@ -6,6 +6,25 @@
 
 #pragma once
 
+#include <filesystem>
+
+#include "outcome/outcome.hpp"
+
 namespace kagome::parachain {
-  int pvf_worker_main(int argc, const char **argv);
-}
+
+  struct SecureModeError {
+    std::string_view message() const {
+      return message_;
+    }
+
+    std::string message_;
+  };
+
+  outcome::result<void, SecureModeError> changeRoot(
+      const std::filesystem::path &worker_dir);
+  outcome::result<void, SecureModeError> enableSeccomp();
+  outcome::result<void, SecureModeError> enableLandlock(
+      const std::filesystem::path &worker_dir);
```

Comment:
Some comments would be helpful. Probably makes sense to reference code from polkadot-sdk